libs/pika/async_cuda/tests/unit/cuda_sender.cu

```diff
@@ -126,8 +125,16 @@ int pika_main(pika::program_options::variables_map& vm)
     std::size_t device = vm["device"].as<std::size_t>();
     //
     unsigned int seed = (unsigned int) std::time(nullptr);
+#if !defined(PIKA_HAVE_HIP)
+    // We undefined this due to a internal compiler error with ROCm Clang-15
+    // triggered when calling the count() method on the variables_map.
     if (vm.count("seed"))
         seed = vm["seed"].as<unsigned int>();
+#else
+    std::cout << "hip Clang is failing at compiling the variables_map methods.";
+    std::cout << " We are therefore ignoring the seed commandline argument ";
+    std::cout << "when hip is enabled." << std::endl;
```

Comment:
Maybe (also apply clang-format here, this line is definitely too long...):
```suggestion
    std::cout << "The --seed command line argument is ignored because HIP is enabled. See https://github.com/pika-org/pika/issues/585 for more details." << std::endl;
```